explorer/src/test/java/com/scalar/client/tool/explorer/ExplorerTest.java

```diff
@@ -44,196 +41,195 @@
+    JsonObject expected = Json.createObjectBuilder()
+            .add("status", StatusCode.OK.toString())
+            .add("history", Json.createArrayBuilder()
+                    .add(Json.createObjectBuilder()
+                            .add("id", "bar")
+                            .add("age", 0)
+                            .add("data", Json.createObjectBuilder()))
+                    .build())
+            .build();
+    when(contractExecutionResultOK.getResult()).thenReturn(Optional.of(expected));
     when(clientService.executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseContractNotFound,
-            contractExecutionResponseOK);
+        .thenReturn(contractExecutionResultOK);
 
     // Act
-    JsonArray result = explorer.scan(ASSET_ID, Json.createObjectBuilder().build());
+    JsonArray actual = explorer.scan(ASSET_ID, Json.createObjectBuilder().build());
 
     // Assert
-    verify(clientService, Mockito.times(3))
-        .executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class));
-    assertThat(result.getJsonObject(0).getString("id")).isEqualTo("bar");
-    assertThat(result.getJsonObject(0).getInt("age")).isEqualTo(0);
+    verify(clientService).executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class));
+    assertThat(actual).isEqualTo(expected.getJsonArray("history"));
   }
 
-  @Test
+  @Test //(expected = ClientException.class)
   public void scan_FailedResponseWhenRegisteringCertificate_ShouldThrowExplorerException() {
     // Arrange
     when(clientService.executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class)))
-        .thenReturn(contractExecutionResponseCertificateNotFound);
-    when(clientService.registerCertificate()).thenReturn(ledgerServiceResponseCertificateNotFound);
+        .thenThrow(clientExceptionCertificateNotFound);
+    doNothing().when(clientService).registerCertificate();
+
+    //Act
+    Throwable thrown = catchThrowable(() -> { explorer.scan(ASSET_ID, Json.createObjectBuilder().build()); });
 
     // Act-assert
-    assertThatThrownBy(() -> explorer.scan(ASSET_ID, Json.createObjectBuilder().build()))
-        .isInstanceOf(ExplorerException.class)
-        .hasMessage("403 certificate not found");
+    assertThat(thrown).isInstanceOf(ClientException.class);
+    ClientException clientException = (ClientException) thrown;
+    assertThat(clientException.getStatusCode()).isEqualTo(StatusCode.CERTIFICATE_NOT_FOUND);
   }
 
   @Test
-  public void scan_CertificateNotFoundTwice_ShouldThrowExplorerException() {
+  public void get_ContractNotFound_ShouldThrowExplorerException() {
+    // Arrange
+    when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
+            .thenThrow(clientExceptionContractNotFound);
+
+    //Act
+    Throwable thrown = catchThrowable(() -> { explorer.get(ASSET_ID); });
+
+    // Act-assert
```

Comment:
```suggestion
    // Assert
```

Reply:
It is a super minor thing, but please fix others.

Reply:
Thank you for review comment.
Review changes updated.